### DISCUSSION TITLE
check if object exists before borrow

### DISCUSSION THREAD

**Discussion by spielcrypto** [aptos-labs/aptos-developer-discussions#366] [Questions] [answered]
### Discord user ID

spielcrypto

### Describe your question in detail.

I have this view function:

```move
#[view]
public fun get_balance_to_claim(account_addr: address): u64 acquires FeesToClaim {
    let fees_to_claim = borrow_global<FeesToClaim>(account_addr);
    fees_to_claim.amount
}
```
I think if `FeesToClaim` object is not created for that account address, it will fail but I don't know how can I check if object exists for that account because I cannot find any borrow_option

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by gregnazario** [comment, marked as answer]
This is a very common pattern with the resource functions. You can use `exists<>`. This does bring a good point though, probably something that could be good to add a `borrow_option` or `borrow_with_default` function

```move
module 0x42::example {
  #[view]
  public fun get_balance_to_claim(account_addr: address): Option<u64> acquires FeesToClaim {
      if (exists<FeesToClaim>(account_addr) {
          let fees_to_claim = borrow_global<FeesToClaim>(account_addr);
          option::some(fees_to_claim.amount)
      } else {
          option::none()
      }
  }
}
```
